tests/utils/splitAtom.test.tsx

```diff
@@ -373,5 +373,64 @@ it('no error on wrong atom configs (fix 510)', async () => {
   fireEvent.click(evenCheckboxEl)
 
   expect(numbersEl.textContent).toBe('0')
-  expect(console.warn).toHaveBeenCalledTimes(1)
+  expect(console.warn).toHaveBeenCalledTimes(0)
+})
+
+it('variable sized splitted atom', async () => {
+  const collectionAtom = atom<number[]>([])
+  const collectionAtomsAtom = splitAtom(collectionAtom)
+
+  const derivativeAtom = atom((get) =>
+    get(collectionAtomsAtom).map(
+      (ca) => get(ca) + Math.round(Math.random() * 150)
+    )
+  )
+
+  const numberAtom = atom(1)
+
+  const generateCollection = (number: number) =>
+    Array.from({ length: Math.round(Math.random() * 30) }, (_, i) => i + number)
+
+  function App() {
+    const [number, setNumber] = useAtom(numberAtom)
+    const setCollection = useUpdateAtom(collectionAtom)
+    const [derivative] = useAtom(derivativeAtom)
+
+    const generatedCollection = generateCollection(number)
+
+    useEffect(() => {
+      const oneArray = [1, 2, 3]
+      const twoArray = [1, 2]
+      if (number === 1) {
+        setCollection(oneArray)
+      } else {
+        setCollection(twoArray)
+      }
+    }, [generatedCollection, number, setCollection])
+
+    return (
+      <div>
+        <button
+          onClick={() => {
+            setNumber((prev) => prev + 1)
+          }}>
+          +{number}
+        </button>
+
+        {derivative.map((d, i) => (
+          <p key={i}>{d}</p>
+        ))}
+      </div>
+    )
+  }
+
+  const { getByText } = render(
+    <Provider>
+      <App />
+    </Provider>
+  )
+
+  fireEvent.click(getByText('+1'))
+
+  expect(console.warn).not.toHaveBeenCalled()
```

Comment:
Is this the only "expect" this will fail? It would be nice if we check `undefined` somehow and remove the warn check.